### PR TITLE
Add Snakefile with workflow rules

### PR DESCRIPTION
**Pull request by pmb59**
This PR adds rules equivalent to those used for GTEX analysis previously, but adapted to how EGA provides the data and metadata.

### CODE REVIEW

File: scripts/aggregate_slurm.sh [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+# Write library paths to file to pass to iRAP for aggregation
+
+aux=${ISL_TEMP_DIR}/aux.${studyId}.${organism}.$$
+echo $libraryPathsForStudy > $aux
+            
+# Submit aggregation to cluster
+
+aggrCmd="irap_single_lib2report_atlas -B -j 4 folders_file=$aux out=$ISL_WORKING_DIR/studies/$studyId/$organism name=$studyId"
+job_name="isl.aggregate.${studyId}.${organism}"
+job_id=$(sbatch -J $job_name --mem $slurmMem -c 1 -e ${job_name}.err -o ${job_name}.out --chdir=$ISL_WORKING_DIR -p production $aggrCmd | awk '{print $4}')
```
Comment (irisdianauy) [inline scripts/aggregate_slurm.sh]: Wrap the command string in a simple shell script with the argument `--wrap`, as sbatch expects a batch script by default.  Time `--time` is also required in our slurm setup. Below is the max time for prod partition.

```suggestion
job_id=$(sbatch --time 7-00:00:00 -J $job_name --mem $slurmMem -c 1 -e ${job_name}.err -o ${job_name}.out --chdir=$ISL_WORKING_DIR -p production --wrap "$aggrCmd" | awk '{print $4}')
```

We can also avoid hard-coding parameters here by importing atlas-bash-util and using atlas-slurm, as in ISL, but perhaps for future implementations.

**Review by irisdianauy — COMMENTED** [review]
Please see comment on the sbatch command to submit the aggregation job.

**Comment by pmb59** [comment]
NOTE on [34e31b2](https://github.com/ebi-gene-expression-group/atlas-ca-analysis/pull/1/commits/34e31b2a949d6fb92f04339ce81adc4091dd240c). `aggregate_slurm_isl.sh` uses  `isl_slurm_submit` [here](https://github.com/ebi-gene-expression-group/isl/blob/618ccc0207562a9283d90a5e39d57ef294be06f1/lib/slurm.sh#L63-L113), and `isl_slurm_monitor` [here](https://github.com/ebi-gene-expression-group/isl/blob/618ccc0207562a9283d90a5e39d57ef294be06f1/lib/slurm.sh#L124-L196).

File: scripts/aggregate_slurm.sh [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+# Write library paths to file to pass to iRAP for aggregation
+
+aux=${ISL_TEMP_DIR}/aux.${studyId}.${organism}.$$
+echo $libraryPathsForStudy > $aux
+            
+# Submit aggregation to cluster
+
+aggrCmd="irap_single_lib2report_atlas -B -j 4 folders_file=$aux out=$ISL_WORKING_DIR/studies/$studyId/$organism name=$studyId"
+job_name="isl.aggregate.${studyId}.${organism}"
+job_id=$(sbatch --time 7-00:00:00 -J $job_name --mem $slurmMem -c 1 -e ${job_name}.err -o ${job_name}.out --chdir=$ISL_WORKING_DIR -p production --wrap $aggrCmd | awk '{print $4}')
```
Comment (irisdianauy) [inline scripts/aggregate_slurm.sh]: Wrap command in quotes
```suggestion
job_id=$(sbatch --time 7-00:00:00 -J $job_name --mem $slurmMem -c 1 -e ${job_name}.err -o ${job_name}.out --chdir=$ISL_WORKING_DIR -p production --wrap="$aggrCmd" | awk '{print $4}')
```

**Review by irisdianauy — APPROVED** [review]
LGTM, apart from the sbatch `--wrap` comment

**Comment by pmb59** [comment]
Thanks for reviewing @irisdianauy

**Pull request merged**